[CI] IndexingShardRelocationIT testPreferredNodeIdsAreUsedDuringRelocation failing

**Build Scans:**
- [elasticsearch-periodic #12315 / concurrent-search-tests](https://gradle-enterprise.elastic.co/s/kwfimvujgt5p2)
- [elasticsearch-pull-request #148909 / part-2](https://gradle-enterprise.elastic.co/s/a6g7vp7tofhuu)
- [elasticsearch-pull-request #148909 / part-2](https://gradle-enterprise.elastic.co/s/wkzmb3qrhiqxq)
- [elasticsearch-pull-request #148906 / part-2](https://gradle-enterprise.elastic.co/s/fzd355frhdb4u)
- [elasticsearch-pull-request #148906 / part-2](https://gradle-enterprise.elastic.co/s/66hqrmohkvo2e)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.recovery.IndexingShardRelocationIT.testPreferredNodeIdsAreUsedDuringRelocation" -Dtests.seed=EF112B0100A870AA -Dtests.jvm.argline="-Des.concurrent_search=true" -Dtests.locale=ar-SY -Dtests.timezone=Antarctica/Rothera -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.recovery.IndexingShardRelocationIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testPreferredNodeIdsAreUsedDuringRelocation'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: safeAwait: CountDownLatch did not reach zero within the timeout
```

**Issue Reasons:**
- [main] 5 failures in test testPreferredNodeIdsAreUsedDuringRelocation (2.9% fail rate in 173 executions)
- [main] 4 failures in step part-2 (4.0% fail rate in 101 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (2.0% fail rate in 99 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -588,9 +588,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
   method: test {csv-spec:approximation.Subqueries with stats by}
   issue: https://github.com/elastic/elasticsearch/issues/149842
-- class: org.elasticsearch.xpack.stateless.recovery.IndexingShardRelocationIT
-  method: testPreferredNodeIdsAreUsedDuringRelocation
-  issue: https://github.com/elastic/elasticsearch/issues/149851
 - class: org.elasticsearch.indices.state.OpenCloseIndexIT
   method: testTranslogStats
   issue: https://github.com/elastic/elasticsearch/issues/149852

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/recovery/IndexingShardRelocationIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/recovery/IndexingShardRelocationIT.java
@@ -885,6 +885,7 @@ public class IndexingShardRelocationIT extends AbstractStatelessPluginIntegTestC
                 .put(IndexSettings.INDEX_TRANSLOG_FLUSH_THRESHOLD_SIZE_SETTING.getKey(), ByteSizeValue.ofGb(1L))
                 .put(IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey(), TimeValue.MINUS_ONE)
                 .put(MaxRetryAllocationDecider.SETTING_ALLOCATION_MAX_RETRY.getKey(), 0)
+                .put(MergePolicyConfig.INDEX_MERGE_ENABLED, false)
                 .build()
         );
         ensureGreen(indexName);

